Employee_Management_Portal/EmployeeManagementPortal/EmployeeManagementPortal/src/main/java/com/portal/validation/Validation.java

```diff
@@ -0,0 +1,206 @@
+     * @throws WrongCredentialsException if the provided email is invalid.
+     */
+    public final boolean checkLoginDto(final LoginInDTO loginDto) {
+        if (!checkName(loginDto.getEmail())) {
+            throw new WrongCredentialsException(
+                    "Please provide a valid email address.");
+        }
+        return true;
+    }
+
+    /**
+     * check if email is already present.
+     * @param email employee email.
+     * @throws Resource already exists exception.
+     */
+    public void checkEmailPresent(final String email) {
+        Optional<Employee> emp = userRepository.findByEmail(email);
+        if (emp.isPresent()) {
+            LOGGER.error("Email id already exists");
+            throw new DuplicateEntryException("Email id already exists");
+        }
+    }
+
+    /**
+     * check if employee id is already present.
+     * @param empId employee id.
+     * @throws Resource already exists exception.
+     */
+    public void checkEmpId(final String empId) {
+        Optional<Employee> emp = userRepository.findByEmpId(empId);
+        if (emp.isPresent()) {
+            LOGGER.error("Employee id already exists");
+            throw new DuplicateEntryException(
+                    "Employee id already exists");
+        }
+    }
+
+    /**
+     * check to perform admin operation.
+     * @param adminDTO Employee dto containing admin informations.
+     */
+    public void checkAdmin(@Valid final EmployeeInDTO adminDTO) {
+        checkEmailPresent(adminDTO.getEmail());
+        checkEmpId(adminDTO.getEmpId());
+    }
+
+    /**
+     * check if email is not present.
+     * @param email employee email.
+     * @throws Resource not found exception.
+     */
+    public void checkEmailEmpty(final String email) {
+        Optional<Employee> emp = userRepository.findByEmail(email);
+        if (emp.isEmpty()) {
+            LOGGER.error("Invalid user");
+            throw new ResourceNotFoundException("Invalid user");
+        }
+    }
+
+    /**
+     * check to perform login operation.
+     * @param loginDTO login dto containing login informations.
+     */
+    public void checkLogin(@Valid final LoginInDTO loginDTO) {
+        checkEmailEmpty(loginDTO.getEmail());
+    }
+
+    /**
+     * check to perform employee operation.
+     * @param empDto employee dto containing employee informations.
+     */
+    public void checkEmployee(@Valid final EmployeeInDTO empDto) {
+        // TODO Auto-generated method stub
+        checkEmailPresent(empDto.getEmail());
+        checkEmpId(empDto.getEmpId());
+    }
+
+    /**
+     * check if project name is already present.
+     * @throws Resource already exists exception.
+     * @param name
+     */
+    public void checkProjectName(final String name) {
+        Project project = projectRepository.findByName(name);
+        if (project != null) {
+        LOGGER.error("Project Name already exists");
+            throw new DuplicateEntryException(
```

Comment:
do not pass string message directly when throwing exceptions at all places . Make another class ('ex: ErrorConstants.java) and store all messages in that class as constants and then call it here.